src/parquet/file/file-metadata-test.cc

```diff
@@ -0,0 +1,178 @@
+
+  std::vector<RowGroupMetaDataWriter*> rowgroups;
+  rowgroups.push_back(rg_writer.get());
+  f_writer->SetRowGroups(rowgroups);
+
+  // schema elements
+  s_writer1->PhysicalType(Type::BOOLEAN);
+  s_writer1->num_children(2);
+  s_writer1->field_id(0);
+  s_writer1->name("schema");
+  s_writer1->ConvertedType(LogicalType::UTF8);
+  s_writer1->RepetitionType(Repetition::REQUIRED);
+
+  s_writer2->PhysicalType(Type::FLOAT);
+  s_writer2->num_children(0);
+  s_writer2->name("float_col");
+  s_writer2->ConvertedType(LogicalType::UTF8);
+  s_writer2->RepetitionType(Repetition::OPTIONAL);
+
+  s_writer3->PhysicalType(Type::INT32);
+  s_writer3->num_children(0);
+  s_writer3->name("int_col");
+  s_writer3->ConvertedType(LogicalType::UTF8);
+  s_writer3->RepetitionType(Repetition::REPEATED);
+
+  std::vector<SchemaMetaDataWriter*> schemas;
+  schemas.push_back(s_writer1.get());
+  schemas.push_back(s_writer2.get());
+  schemas.push_back(s_writer3.get());
+  f_writer->SetSchemas(schemas);
+
+  std::vector<uint8_t> metadata_bytes(f_writer->get_buffer_size());
+  memcpy(metadata_bytes.data(), f_writer->get_buffer(), f_writer->get_buffer_size());
+
+  // Read the metadata
+  std::unique_ptr<FileMetaDataReader> f_reader =
+      FileMetaDataReader::GetFileMetaDataReader(metadata_bytes.data());
+  // file metadata
+  ASSERT_EQ(nrows, f_reader->num_rows());
+  ASSERT_EQ(1, f_reader->num_row_groups());
+  ASSERT_EQ(DEFAULT_WRITER_VERSION, f_reader->version());
+  ASSERT_EQ(DEFAULT_CREATED_BY, f_reader->created_by());
+  ASSERT_EQ(3, f_reader->num_schema_elements());
+  // row group metadata
+  std::unique_ptr<RowGroupMetaDataReader> rg_reader =
+      f_reader->GetRowGroupMetaDataReader(0);
+  ASSERT_EQ(2, rg_reader->num_columns());
+  ASSERT_EQ(nrows, rg_reader->num_rows());
+  ASSERT_EQ(1024, rg_reader->total_byte_size());
+  ASSERT_EQ(true, rg_reader->IsColumnStatsSet(0));
+  ASSERT_EQ(true, rg_reader->IsColumnStatsSet(1));
+  ASSERT_EQ("100.100", *rg_reader->GetColumnStats(0).min);
+  ASSERT_EQ("200.200", *rg_reader->GetColumnStats(0).max);
+  ASSERT_EQ("100", *rg_reader->GetColumnStats(1).min);
+  ASSERT_EQ("200", *rg_reader->GetColumnStats(1).max);
+  ASSERT_EQ(0, rg_reader->GetColumnStats(0).null_count);
+  ASSERT_EQ(0, rg_reader->GetColumnStats(1).null_count);
+  ASSERT_EQ(nrows, rg_reader->GetColumnStats(0).distinct_count);
+  ASSERT_EQ(nrows, rg_reader->GetColumnStats(1).distinct_count);
+  ASSERT_EQ(DEFAULT_COMPRESSION_TYPE, rg_reader->GetColumnCompression(0));
+  ASSERT_EQ(DEFAULT_COMPRESSION_TYPE, rg_reader->GetColumnCompression(1));
+  ASSERT_EQ(DEFAULT_ENCODING, rg_reader->GetColumnEncodings(0)[2]);
+  ASSERT_EQ(DEFAULT_ENCODING, rg_reader->GetColumnEncodings(1)[2]);
+  ASSERT_EQ(512, rg_reader->GetColumnCompressedSize(0));
+  ASSERT_EQ(512, rg_reader->GetColumnCompressedSize(1));
+  ASSERT_EQ(512, rg_reader->GetColumnUnCompressedSize(0));
+  ASSERT_EQ(512, rg_reader->GetColumnUnCompressedSize(1));
```

Comment:
Same question as elsewhere whether always referring to `ColumnMetaData` by index or adding a separate `parquet::ColumnMetaData` accessor object